dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,63 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::IndexSelect(
+    int64_t dim, torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  if ((!dim && HasCSR()) || (dim && HasCSC()) || !HasCOO()) {
```

Comment:
Define `bool rowwise = dim==0` in advance to make the later `if` conditions easier to understand.

Reply:
Better to define the used sparse format earlier to simply this if condition.

```cpp
SparseFormat sparse_format = ...
if (sparse_format == SparseFormat::kCSR || sparse_format == SparseFormat::kCSC) {
...
}
```

Reply:
The sparse format may contain different formats at the same time, I have no idea whether sparse_format is suitable.

Reply:
I mean you can simplify the code here. This if condition is not readable. You change the code like this
```cpp
SparseFormat sparse_format;
if (rowwise) {
  if (HasCSR() || !HasCOO()) {
    sparse_format = SparseFormat::kCSR;
  }
  else {
    sparse_format = SparseFormat::kCOO;
  }
} else {
  ...
}
```

We may have a utility function for deciding the format later.